spec/integration/type/file_spec.rb

```diff
@@ -93,6 +93,43 @@ def get_aces_for_path_by_sid(path, sid)
     expect(Puppet::FileSystem.exist?(source)).to be_falsey
   end
 
+  describe "when ensure is present using an empty file" do
+    before(:each) do
+      catalog.add_resource(described_class.new(:path => path, :ensure => :present, :backup => :false))
+    end
+
+    context "file is present" do
+      before(:each) do
+        FileUtils.touch(path)
+      end
+
+      it "should do nothing" do
+        report = catalog.apply.report
+        expect(report.resource_statuses["File[#{path}]"]).not_to be_failed
+        expect(Puppet::FileSystem.exist?(path)).to be_truthy
+      end
+
+      it "should log nothing" do
+        logs = catalog.apply.report.logs
+        expect(logs).to be_empty
+      end
+    end
+
+    context "file is not present" do
+      it "should create the file" do
+        report = catalog.apply.report
+        expect(report.resource_statuses["File[#{path}]"]).not_to be_failed
+        expect(Puppet::FileSystem.exist?(path)).to be_truthy
+      end
+
+      it "should log that the file was created" do
+        logs = catalog.apply.report.logs
+        expect(logs.first.source).to eq("/File[#{path}]/ensure")
+        expect(logs.first.message).to eq("created")
```

Comment:
@elyscape your comment here might have gotten blown away by my rebase (or you just deleted it:)). If no content is specified, it looks like we still just log "created" with your change:

    Notice: /Stage[main]/Main/File[/tmp/test1]/ensure: created

And then of course, `/defined content.../` when we provide `content => "foo"`:

    Notice: /Stage[main]/Main/File[/tmp/test2]/ensure: defined content as '{md5}877639120b17f7a4dc07783b667f13cf'

Does that all sound correct to you and for this test?